expression/builtin_string_vec.go

```diff
@@ -688,11 +688,55 @@ func (b *builtinCharSig) vecEvalString(input *chunk.Chunk, result *chunk.Column)
 }
 
 func (b *builtinReplaceSig) vectorized() bool {
-	return false
+	return true
 }
 
+// evalString evals a builtinReplaceSig.
+// See https://dev.mysql.com/doc/refman/5.7/en/string-functions.html#function_replace
 func (b *builtinReplaceSig) vecEvalString(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETString, n)
```

Comment:
Can we directly use the result variable to store the result of args[0]? Thus we can improve the performance by saving a memory copy.